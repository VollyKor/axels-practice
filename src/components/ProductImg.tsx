import Image from 'react-bootstrap/Image';

import { Wrapper } from 'styled/utils/common';

import Img from 'images/image-not-found.jpg';

interface Props {
    img?: string;
}

const ProductImg = ({ img }: Props) => (
    <Wrapper className="wrapper">
        <Image
            src={img ?? Img}
            alt=""
            width="50px"
            height="50px"
            className="p-1"
            rounded
        />
    </Wrapper>
);

export default ProductImg;
