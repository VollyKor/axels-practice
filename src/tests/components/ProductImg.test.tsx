import { shallow } from 'enzyme';
import Image from 'react-bootstrap/Image';

import ProductImg from 'components/ProductImg';

import Img from 'images/image-not-found.jpg';

it('renders snapshot', async () => {
    const wrapper = shallow(<ProductImg img="" />);
    expect(wrapper).toMatchSnapshot();
});

it('renders src of image correctly', async () => {
    const someSrc = 'http://img.com';

    const wrapper = shallow(<ProductImg img={someSrc} />);

    const imgSrc = wrapper.find(Image).prop('src');

    expect(imgSrc).toEqual(someSrc);
});

it('renders default image correctly', async () => {
    const wrapper = shallow(<ProductImg />);

    const imgSrc = wrapper.find(Image).prop('src');

    expect(imgSrc).toEqual(Img);
});
