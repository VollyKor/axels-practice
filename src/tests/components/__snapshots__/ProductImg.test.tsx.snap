// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<styled.div
  className="wrapper"
>
  <Image
    alt=""
    className="p-1"
    fluid={false}
    height="50px"
    rounded={true}
    roundedCircle={false}
    src=""
    thumbnail={false}
    width="50px"
  />
</styled.div>
`;
